cpp/inc/bond/core/traits.h

```diff
@@ -193,7 +193,7 @@ is_reader<Input, T, typename boost::enable_if<is_class<typename Input::Parser> >
 template <typename T> struct
 buffer_magic
 {
-    BOOST_STATIC_ASSERT_MSG(!is_same<T, T>::value, "Undefined buffer.");
+    BOOST_STATIC_ASSERT_MSG((!is_same<T, T>::value), "Undefined buffer.");
```

Comment:
Maybe we should add `template <typename T> struct always_false : std::false_type {};` and use as `always_false<T>::value` which will not have `,` in the macro and will be cleaner?

Reply:
Addressed